libs/libc/stdio/lib_getdelim.c

```diff
@@ -235,6 +266,7 @@ ssize_t getdelim(FAR char **lineptr, size_t *n, int delimiter,
 #ifdef HAVE_GETLINE
 ssize_t getline(FAR char **lineptr, size_t *n, FAR FILE *stream)
 {
-  return getdelim(lineptr, n, EOLCH, stream);
+  return getdelimx(lineptr, n, EOLCH, stream,
+    stream == stdin ? stdout : NULL);
```

Comment:
how to handle redirect? the echo should be handled by serial driver, I think.

Reply:
Yes, after reading the docs of termios, the serial driver should handle this behavior (configurable by termios).